File: fibonacci/fibonacci.kt
import java.math.BigInteger

fun fibonacci(n: Int): BigInteger {
    var a: BigInteger = BigInteger.ZERO
    var b: BigInteger = BigInteger.ONE
    for (i in 0 until n) {
        val temp = a
        a = b
        b = b.add(temp)
    }
    return a
}

fun main() {
    val result = fibonacci(100000)
    println(result)
}

File: fibonacci/fibonacci.c
#include <stdio.h>
#include <stdint.h>

uint64_t fibonacci(int32_t n) {
    uint64_t a = 0, b = 1;
    for (int32_t i = 0; i < n; i++) {
        uint64_t temp = a;
        a = b;
        b += temp;
    }
    return a;
}

int main() {
    uint64_t result = fibonacci(100000);
    printf("%llu\n", result);
    return 0;
}